### ISSUE TITLE
Lambda Layer Builder Pulling libraries from 'main' branch when using older tagged version

### ISSUE THREAD

**Issue by rjclarke7** [aws-ia/terraform-aws-control_tower_account_factory#198]
**Terraform Version & Prov: 0.15.5 **

**AFT Version: 1.3.4**
(Can be found in the AFT Management Account in the SSM Parameter `/aft/config/aft/version`)

**Terraform Version & Provider Versions**
Please provide the outputs of `terraform version` and `terraform providers` from within your AFT environment

`terraform version`
```
$ terraform version
Terraform v1.2.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v3.74.3
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
```

`terraform providers`
```
$ terraform providers

Providers required by configuration:
.
└── module.aft
    ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_code_repositories
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_customizations
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_iam_roles
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   ├── module.audit_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    │   ├── module.ct_management_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    │   ├── module.log_archive_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    │   └── module.aft_exec_role
    │       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
    ├── module.aft_feature_options
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.aft_lambda_layer
    │   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    │   ├── provider[registry.terraform.io/hashicorp/random]
    │   └── provider[registry.terraform.io/hashicorp/local]
    ├── module.aft_account_provisioning_framework
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.aft_account_request_framework
    │   ├── provider[registry.terraform.io/hashicorp/time]
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    ├── module.packaging
    │   └── provider[registry.terraform.io/hashicorp/archive]
    ├── module.aft_ssm_parameters
    │   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    └── module.aft_backend
        └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0

Providers required by state:

    provider[registry.terraform.io/hashicorp/local]

    provider[registry.terraform.io/hashicorp/time]

    provider[registry.terraform.io/hashicorp/aws]

    provider[registry.terraform.io/hashicorp/archive]

    provider[registry.terraform.io/hashicorp/random]
```

**Bug Description**
When deploying AFT tag 1.3.4 we noticed no accounts were being vended and diagnosed the issue to continual failures of the aft-account-request-processor Lambda, with repeated log failures of the following message.

```
[ERROR] AttributeError: module 'aft_common.aft_utils' has no attribute 'get_ct_management_session'
Traceback (most recent call last):
  File "/var/task/aft_account_request_processor.py", line 27, in lambda_handler
    ct_management_session = utils.get_ct_management_session(session)
```

I downloaded the aft-common-1-3-4 Lambda layer, unzipped it and found that the aft_utils.py was not from tag 1.3.4, but the latest commit from main, where the attribute 'get_ct_management_session' had been removed, hence was not available.

I determined that during the build of the Lambda layer, the SSM parameter '/aft/config/aft-pipeline-code-source/repo-git-ref' was used to determine the built version of aft_common, where aft_utils.py resides. In our instance, this parameter was 'main', not the tag we had selected during the install.

**To Reproduce**
1. Install any tagged version of AFT before the 'get_ct_management_session' parameter was removed (From 1.3.6 and older).
2. Attempt to vend an account.
3. Review logs for Lambda function aft-account-request-processor.

**Expected behavior**
I had expected that the Lambda layer build parameter should refer to the tag we are using, not the 'main' branch, hence all aft_common library reflects the same version of AFT that matches the rest of the installation.

**Related Logs**
[Container] 2022/06/29 11:08:25 Waiting for agent ping
[Container] 2022/06/29 11:08:26 Waiting for DOWNLOAD_SOURCE
[Container] 2022/06/29 11:08:27 Phase is DOWNLOAD_SOURCE
[Container] 2022/06/29 11:08:27 CODEBUILD_SRC_DIR=/codebuild/output/src374316130/src
[Container] 2022/06/29 11:08:27 YAML location is /codebuild/readonly/buildspec.yml
[Container] 2022/06/29 11:08:27 Processing environment variables
[Container] 2022/06/29 11:08:27 Selecting 'python' runtime version '3.8' based on manual selections...
[Container] 2022/06/29 11:08:27 Running command echo "Installing Python version 3.8 ..."
Installing Python version 3.8 ...

[Container] 2022/06/29 11:08:27 Running command pyenv global  $PYTHON_38_VERSION

[Container] 2022/06/29 11:08:30 Moving to directory /codebuild/output/src374316130/src
[Container] 2022/06/29 11:08:30 Configuring ssm agent with target id: codebuild:767c9666-2c0e-4698-82d7-bca17f29ac17
[Container] 2022/06/29 11:08:30 Successfully updated ssm agent configuration
[Container] 2022/06/29 11:08:30 Registering with agent
[Container] 2022/06/29 11:08:30 Phases found in YAML: 2
[Container] 2022/06/29 11:08:30  INSTALL: 13 commands
[Container] 2022/06/29 11:08:30  BUILD: 5 commands
[Container] 2022/06/29 11:08:30 Phase complete: DOWNLOAD_SOURCE State: SUCCEEDED
[Container] 2022/06/29 11:08:30 Phase context status code:  Message: 
[Container] 2022/06/29 11:08:30 Entering phase INSTALL
[Container] 2022/06/29 11:08:30 Running command DEFAULT_PATH=$(pwd)

[Container] 2022/06/29 11:08:30 Running command AWS_MODULE_SOURCE=$(aws ssm get-parameter --name $SSM_AWS_MODULE_SOURCE --query "Parameter.Value" --output text)

[Container] 2022/06/29 11:08:41 Running command AWS_MODULE_GIT_REF=$(aws ssm get-parameter --name $SSM_AWS_MODULE_GIT_REF --query "Parameter.Value" --output text)

[Container] 2022/06/29 11:08:42 Running command URL=$(echo "$AWS_MODULE_SOURCE" | awk '{split($0,a,"@"); print a[2]}')

[Container] 2022/06/29 11:08:42 Running command ssh_key_parameter=$(aws ssm get-parameter --name /aft/config/aft-ssh-key --with-decryption 2> /dev/null || echo "None")
if [[ $ssh_key_parameter != "None" ]]; then
  ssh_key=$(jq --raw-output ".Parameter.Value" <<< $ssh_key_parameter)
  mkdir -p ~/.ssh
  echo "Host *" >> ~/.ssh/config
  echo "StrictHostKeyChecking no" >> ~/.ssh/config
  echo "UserKnownHostsFile=/dev/null" >> ~/.ssh/config
  echo "$ssh_key" > ~/.ssh/ssh_key
  echo -e "\n\n" >>  ~/.ssh/ssh_key
  chmod 600 ~/.ssh/ssh_key
  eval "$(ssh-agent -s)"
  ssh-add ~/.ssh/ssh_key
fi

[Container] 2022/06/29 11:08:42 Running command git config --global credential.helper '!aws codecommit credential-helper $@'

[Container] 2022/06/29 11:08:43 Running command git config --global credential.UseHttpPath true

[Container] 2022/06/29 11:08:43 Running command echo "Building aft_common from ${URL}:${AWS_MODULE_GIT_REF}"
Building aft_common from :main

[Container] 2022/06/29 11:08:43 Running command git clone -b $AWS_MODULE_GIT_REF $AWS_MODULE_SOURCE aws-aft-core-framework
Cloning into 'aws-aft-core-framework'...

[Container] 2022/06/29 11:08:44 Running command python3 -m pip install virtualenv
Requirement already satisfied: virtualenv in /root/.pyenv/versions/3.8.13/lib/python3.8/site-packages (20.14.1)
Requirement already satisfied: filelock<4,>=3.2 in /root/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from virtualenv) (3.7.1)
Requirement already satisfied: six<2,>=1.9.0 in /root/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from virtualenv) (1.16.0)
Requirement already satisfied: distlib<1,>=0.3.1 in /root/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from virtualenv) (0.3.4)
Requirement already satisfied: platformdirs<3,>=2 in /root/.pyenv/versions/3.8.13/lib/python3.8/site-packages (from virtualenv) (2.5.2)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[Container] 2022/06/29 11:08:50 Running command python3 -m venv .venv

[Container] 2022/06/29 11:08:54 Running command . .venv/bin/activate
.......

**Comment by stumins** [comment]
Hey @rjclarke7, thanks for reporting this issue.

This bug was [resolved in release 1.3.6](https://github.com/aws-ia/terraform-aws-control_tower_account_factory/releases/tag/1.3.6). Versions 1.3.6 and later honor the deployed version when [building the lambda layer](https://github.com/aws-ia/terraform-aws-control_tower_account_factory/blob/main/modules/aft-lambda-layer/buildspecs/aft-lambda-layer.yml#L33). 

Upgrading to 1.3.6 or later should resolve the issue.

**Comment by balltrev** [comment]
Closing this issue in favor of the pinned one, https://github.com/aws-ia/terraform-aws-control_tower_account_factory/issues/205
